resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -168,24 +168,34 @@ private[spark] class Client(
     val initContainerConfigMap = initContainerComponentsProvider
         .provideInitContainerConfigMapBuilder(maybeSubmittedResourceIdentifiers.map(_.ids()))
         .build()
-    val podWithInitContainer = initContainerComponentsProvider
-        .provideInitContainerBootstrap()
-        .bootstrapInitContainerAndVolumes(driverContainer.getName, basePod)
 
     val containerLocalizedFilesResolver = initContainerComponentsProvider
         .provideContainerLocalizedFilesResolver()
     val resolvedSparkJars = containerLocalizedFilesResolver.resolveSubmittedSparkJars()
     val resolvedSparkFiles = containerLocalizedFilesResolver.resolveSubmittedSparkFiles()
-
-    val executorInitContainerConfiguration = initContainerComponentsProvider
+    var needInitContainer = false
+    for (e <- (resolvedSparkJars ++ resolvedSparkFiles)) {
+      if (e.nonEmpty && !e.startsWith("local://")) {
+        needInitContainer = true
+      }
+    }
+    var podBuilder = basePod
```

Comment:
Write the logic such that this can be kept as a `val`. There's a few options, but here's one:

`val (sparkConfWithExecutorInit, podBuilderWithInitContainer) = if (needInitContainer { ... } else (sparkConf, basePod)`